sdl-core/src/main/scala/io/smartdatalake/workflow/dataframe/GenericDataFrame.scala

```diff
@@ -231,7 +231,7 @@ trait GenericSchema extends GenericTypedObject {
     SchemaConverter.convert(this, toSubFeedType)
   }
   def equalsSchema(schema: GenericSchema): Boolean = {
-    diffSchema(schema).nonEmpty || schema.diffSchema(this).nonEmpty
+    diffSchema(schema).isEmpty && schema.diffSchema(this).isEmpty
```

Comment:
looks like a massive bugfix! How come it seemded to have no big impact until now?

Reply:
You're right.... for now this method is only used in IcebergTableDataObject. I did some tests and found this bug.